core/data/src/main/java/com/serranoie/itinero/core/data/remote/BaseApiClient.kt

```
@@ -0,0 +1,111 @@
+            if (body != null) {
+                setBody(body)
+            }
+        }
+        return handleResponse(response)
+    }
+
+    protected suspend inline fun <reified T> handleResponse(response: HttpResponse): T {
+        if (!response.status.isSuccess()) {
+            val errorBody = try {
+                response.body<String>()
+            } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)